src/app/channels/channels.component.ts

```diff
@@ -32,7 +32,7 @@ export class ChannelsComponent implements OnInit {
     private route: ActivatedRoute,
     private channelPermissions: ChannelPermissionsService) { }
 
-  ngOnInit() {
+  ngOnChanges() {
```

Comment:
Dangerous operation!
Don't subscribe inside ngOnChanges.
If you really have to, make sure you unsubscribe to all previous subscriptions.

In this case I don't think that we need the search option in this component so this function can be changed back to ngOnInit instead.

Reply:
Removed channel-search, so no need for subscribe in here anymore.